### PR TITLE
Added Visual studio support.

### PR DESCRIPTION
**Pull request by Thar0l**
Added Visual Studio solution for following projects:
- switched
- subroutined
- predecoded
- tailrecursive
- native

Also added batch srcripts for sanity testing and measuring performance.

### CODE REVIEW

File: native.vcxproj [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  <ImportGroup Label="PropertySheets" Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+  </ImportGroup>
+  <PropertyGroup Label="UserMacros" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
+    <LinkIncremental>true</LinkIncremental>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
+    <LinkIncremental>true</LinkIncremental>
+    <LibraryPath>C:\Users\Алексей\Documents\Visual Studio 2015\Projects\interpreters-comparison\x64\Debug;$(LibraryPath)</LibraryPath>
```
Comment (grigory-rechistov) [inline native.vcxproj]: There are hardcoded paths in your configuration:

`<LibraryPath>C:\Users\Алексей\Documents\Visual Studio 2015...`

I cannot build from these project files (**common.lib** cannot be found) unless I manually fix all the references in all projects in all configurations.

Instead, environment variables should be used. At the build time, they are substituted to "right" paths on each host. For example:

`$(MSBuildProjectDirectory)\x64\Debug`

With this library path, I managed to build a project

**Comment by Thar0l** [comment]
Fixed

**Comment by grigory-rechistov** [comment]
`tailrecursive` variant crashes with stack overflow both in Debug and Release modes. Apparently, tail-call optimization was not applied. Can you please try to make it work?

**Comment by grigory-rechistov** [comment]
I tried it with VS 2015 Update 1, x64 builds

**Comment by Thar0l** [comment]
There is a problem with founding alternative for -f-optimize-sibling-calls in VS. Max optimization with /O2 and /Oy or  /OG doeen't work. I'll try to find out something, but one stupid solution is to build with /STACK 100000000. But it's a glitch.

**Comment by grigory-rechistov** [comment]
If nothing helps, then disable the `tailrecursive` build altogether.

**Comment by Thar0l** [comment]
Disabled.

**Pull request merged**